src/components/Expandable.js

```diff
@@ -0,0 +1,45 @@
+import { html } from 'htm/preact';
+import { useState, useRef, useEffect } from 'preact/compat';
+
+const chevronRight = new URL('../assets/icons/chevron-right.svg', import.meta.url).href;
+const chevronDown = new URL('../assets/icons/chevron-down.svg', import.meta.url).href;
+
+export function Expandable(props) {
+  const [expanded, setExpanded] = useState(false);
+  const detail = useRef(null);
+
+  useEffect(() => {
+    if (expanded) {
+      detail.current.focus();
+    }
+  }, [expanded]);
+
+  return html`
+    <div class="ld-expandable">
+      <button
+        onClick=${() => setExpanded(!expanded)}
+        id="button1"
+        class="ld-expandable--toggle"
+        aria-controls="t1"
+        aria-expanded="${expanded}"
+      >
+        <div class="ld-expandable--icon">
+          <img src=${expanded ? chevronDown : chevronRight} alt="" />
+        </div>
+        <div class="ld-expandable--toggle-content">
+          <h2>${props.toggle}</h2>
+        </div>
+      </button>
+
+      <div
+        ref=${detail}
+        id="t1"
+        class="ld-expandable--detail ${expanded ? 'ld-expandable--expanded' : 'ld-expandable--closed'}"
+        role="region"
+        tabindex="-1"
+      >
+        ${props.detail}
```

Comment:
You can use a slot for this, and get the slotted content with the `children` property

Reply:
I considered using `props.children`, but that would not account for if someone does:

```js
<${Expandable}>
 <h2>foo</h2>
 <p>some text</p>
 <p>more text</p>
<//>
```

so i figured i'd make it more explicit and use render props instead

Reply:
hmm, I expected that would work with multiple children?

Reply:
well yeah you can, but props.children is an array so you have to do `props.children[0]` etc, with a renderprop its more clear/explicit what you want to render where

Reply:
I'm not sure i understand, doesn't preact render an array? Or do you want to ensure there is only one node?